src/hasher.cpp

```diff
@@ -213,7 +213,8 @@ TORRENT_CRYPTO_NAMESPACE
 #elif TORRENT_USE_CRYPTOAPI
 		m_context.update(data);
 #elif defined TORRENT_USE_LIBCRYPTO
-		SHA256_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), data.size());
+		SHA256_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data())
+			, static_cast<std::size_t>(data.size()));
```

Comment:
I'm not sure why this didn't warn before. I'm tempted to put this in master as a separate commit, ahead of v2